### ISSUE TITLE
TF-slim resume training

### ISSUE THREAD

**Issue by rickymf4** [tensorflow/models#975]
## slim/train_image_classifier.py

As https://github.com/tensorflow/models/tree/master/slim described, the TF-slim noly can Training from scratch and Fine tuning to a new task . 
Because I may halt training and change learning-rate, and then resume the training. Is there any way to resume training from a checkpoint, base on TF-slim ?   

Thanks!

**Comment by ogail** [comment]
Yes, provide the train directory as the checkpoint directory and TF-Slim will pick up the latest checkpoint automatically.

**Comment by rickymf4** [comment]
Thanks @ogail ! However, when you provide the train directory as the checkpoint directory, you have to provide another directory for the train directory, because the following code:

` 

    if tf.train.latest_checkpoint(FLAGS.train_dir):
        tf.logging.info(
            'Ignoring --checkpoint_path because a checkpoint already exists in %s'
             % FLAGS.train_dir)
        return None
`

So the training status such as `global step` will lost, and the train will start from step 0 and read the TFRecords from beginning again.

**Comment by ogail** [comment]
@rickymf4 if the TRAIN_DIR has `checkpoint` file, then TF will use the latest one specified by that file automatically and you don't need to provide anything (actually remove the --checkpoint_path). In case your checkpoint is saved somewhere else (which is unlikely as you need to remove `checkpoint` file) then pass --checkpoint_path parameter that points out to the model.ckpt file.

FYI, I do same of your use case by:

- Train model for one epoch
- Evaluate training and validation sets
- Check if learning rate should decay based on the evaluation numbers or not
- Repeat, from start.

What I mentioned above should work, check out the training script I have [here](https://github.com/ogail/models/blob/master/slim/scripts/train_experiment.sh#L166).

**Comment by sabhiram** [comment]
What about the data from a dataset provider? Does the checkpoint track the offset of the batch provider to ensure the same input data is not being fed into the training stream again?

**Comment by karmel** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
